pkg/helm/manifest.go

```diff
@@ -36,6 +36,25 @@ type ManifestObjectFilter struct {
 	Labels map[string]string
 }
 
+type ObjectList struct {
+	ObjectKey
+	*ManifestObject
+}
+
```

Comment:
How about introducing an Interface that only requires the `Matches(obj *resource.Info) bool`
```golang
type ManifestFilter interface {
	Matches(obj *resource.Info) bool
}
```

Then it is not necessary to create the additional functions to filter for release/manifest per filter, but the existing methods can be refactored to accept this interface instead

```golang
func ObjectMapFromRelease(restClientGetter genericclioptions.RESTClientGetter, r *release.Release, f ManifestFilter) (map[ObjectKey]*ManifestObject, error)

&&

func ObjectMapFromManifest(restClientGetter genericclioptions.RESTClientGetter, namespace, manifest string, f ManifestFilter) (map[ObjectKey]*ManifestObject, error) 
```

Then it will be possible to add more structs that implement the interface, also for reverse filtering. Only difference is currently the return of the new struct `ObjectList` instead of `map[ObjectKey]*ManifestObject`. The calling code could also work with the map instead.

Reply:
done in [0dbf203](https://github.com/cloudoperators/greenhouse/pull/398/commits/0dbf203745e991ce5aa7bee76209620bee38e49e)